pkg/identity/identity.go

```diff
@@ -76,3 +79,87 @@ func (id *signingIdentity) Credentials() []byte {
 func (id *signingIdentity) Sign(message []byte) ([]byte, error) {
 	return id.sign(message)
 }
+
+func CreateSigner(privateKeyPath, certificatePath, mspID string) (SigningIdentity, error) {
+	priv, err := GetPrivateKey(privateKeyPath)
+	if err != nil {
+		return nil, err
+	}
+
+	cert, _, err := GetCertificate(certificatePath)
+	if err != nil {
+		return nil, err
+	}
+
+	return NewPrivateKeySigningIdentity(mspID, cert, priv)
+}
+
+func GetPrivateKey(f string) (*ecdsa.PrivateKey, error) {
+	in, err := os.ReadFile(f)
+	if err != nil {
+		return nil, err
+	}
+
+	k, err := PEMtoPrivateKey(in)
+	if err != nil {
+		return nil, err
+	}
+
+	key, ok := k.(*ecdsa.PrivateKey)
+	if !ok {
+		return nil, errors.New("expecting ecdsa key")
+	}
+
+	return key, nil
+}
+
+func GetCertificate(f string) (*x509.Certificate, []byte, error) {
+	in, err := os.ReadFile(f)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	block, _ := pem.Decode(in)
+
+	c, err := x509.ParseCertificate(block.Bytes)
+	return c, in, err
+}
+
+// PEMtoPrivateKey unmarshals a pem to private key
+func PEMtoPrivateKey(raw []byte) (interface{}, error) {
+	if len(raw) == 0 {
+		return nil, errors.New("invalid PEM. It must be different from nil")
+	}
+	block, _ := pem.Decode(raw)
+	if block == nil {
+		return nil, fmt.Errorf("failed decoding PEM. Block must be different from nil. [% x]", raw)
+	}
+	cert, err := DERToPrivateKey(block.Bytes)
+	if err != nil {
+		return nil, err
+	}
+	return cert, err
+}
+
+// DERToPrivateKey unmarshals a der to private key
+func DERToPrivateKey(der []byte) (key interface{}, err error) {
+
+	if key, err = x509.ParsePKCS1PrivateKey(der); err == nil {
+		return key, nil
+	}
+
+	if key, err = x509.ParsePKCS8PrivateKey(der); err == nil {
+		switch key.(type) {
+		case *ecdsa.PrivateKey:
+			return
+		default:
+			return nil, errors.New("found unknown private key type in PKCS#8 wrapping")
+		}
+	}
+
+	if key, err = x509.ParseECPrivateKey(der); err == nil {
+		return
```

Comment:
For readability, it is a bit odd that only this return point uses the named return values whereas all the others explicitly state the return values